code/game/atoms_movable.dm

```diff
@@ -1024,6 +1024,10 @@
 /atom/movable/proc/do_attack_animation(atom/target, atom/tool)
 	set waitfor = 0
 
+	if(tool.flags&NO_ATTACK_ANIM)
```

Comment:
Put this after the ASSERT. It doesn't really matter: if `tool` is null this will also runtime, but I guess the error message will be clearer.

Reply:
fixed